### DISCUSSION TITLE
Creating new tags for select and multi-select columns via API [feature request]

### DISCUSSION THREAD

**Discussion by RawmanGrey** [nocodb/nocodb#3663] [General]
If POST request contains new tags targeting select or multi-select values, those text values are set and returned in response but no tags are created and shown in NocoDB UI.

<details> 
  <summary>POST request+response screenshoot </summary>
   
![Screenshot_20220915_121607](https://user-images.githubusercontent.com/61871501/190367966-7690e32c-f24e-48c2-ac24-3d14000beb40.jpg)

</details>

<details> 
  <summary> gif showing result in UI</summary>
   
![Peek 2022-09-15 12-27](https://user-images.githubusercontent.com/61871501/190368597-5eb6708f-d0f8-41e7-a562-bd6c67ebefba.gif)

   
</details>

Feels related to #3512 

Node: **v12.22.12**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.96.4**

**Comment by o1lab** [comment]
You would have to create new tags and then insert it. We WONT automatically create tags for you - although this is a nice feature when doing it from the UI :)

**Comment by RawmanGrey** [comment]
> You would have to create new tags and then insert it. We WONT automatically create tags for you - although this is a nice feature when doing it from the UI :)

Yeah, fair enough -- let it be a feature request : )

Though, at least some non-handwork way to create plenty of tags has to be... I made this issue mostly to comment on mentioned csv-import request.

**Comment by o1lab** [comment]
Thank you :)

Will move this to discussion.

**Comment by shkomg** [comment]
@RawmanGrey @o1lab 
please explain or I missing something:
1. there is an api to create column - https://meta-apis-v2.nocodb.com/#tag/DB-Table-Column/operation/db-table-column-create
2. there is an api to create record - https://data-apis-v2.nocodb.com/#tag/Table-Records/operation/db-data-table-row-create

But if I want to create SingleSelect or MultiSelect where is the api to add tags for the column to select from?

It works fine when I upload csv - let's to make a set of tags from the csv values.
It works fine when I add value manually over UI - new tag gets created just fine.

**But how may I create table with SingleSelect / MultiSelect  columns and tags sets using API ????**

**Reply by EugeneKonovalov** [reply]
> @RawmanGrey @o1lab please explain or I missing something:
> 
>     1. there is an api to create column - https://meta-apis-v2.nocodb.com/#tag/DB-Table-Column/operation/db-table-column-create
> 
>     2. there is an api to create record - https://data-apis-v2.nocodb.com/#tag/Table-Records/operation/db-data-table-row-create
> 
> 
> But if I want to create SingleSelect or MultiSelect where is the api to add tags for the column to select from?
> 
> It works fine when I upload csv - let's to make a set of tags from the csv values. It works fine when I add value manually over UI - new tag gets created just fine.
> 
> **But how may I create table with SingleSelect / MultiSelect columns and tags sets using API ????**

I agree! I have same problem, can't find any api possibilities to create tags for selects, so it means that it's impossible to create functional select or multi-select column via api... :(

**Reply by sant1dom** [reply]
You can do it programmatically. 
Retrieve the details of the column using the API, then make a PATCH request to update the column with the same data structure but append the option object you want at the end of the options array.
To make it work you need to specify just the color and the title.
Hope it helps.
